### PR TITLE
Comprehensive e2e test suite (Layer 1 + Layer 2)

### PR DESCRIPTION
**Pull request by brotatotes**
## Comprehensive e2e test suite (Layer 1 + Layer 2)

Replaces the old SHL-only, hardcoded `App.test.tsx` with a two-layer puppeteer suite covering **both songbooks (SHL + SFOG)** across **both form factors (mobile ~390px / desktop ~1366px)**, using **live data** fetched at runtime from the Resources repo (no committed snapshot).

### Layer 1 — sampled data checks (`src/test/e2e/Layer1.data.test.tsx`)
A curated representative sample per book (~15–20 songs each, **not** all 813):
- Lyric view renders (`<n>) <title>` + ≥1 lyric line; author shown iff non-empty)
- Music image loads **HTTP 200 / `image/png`** (not HTML) — guards #211
- Presentation-order songs render verses in declared order
- Out-of-range edges (0 and N+1) document current behavior (#147)
- Includes the SHL second-tune songs (156/216/278/478)

### Layer 2 — feature matrix (`src/test/e2e/Layer2.features.test.tsx`)
Each feature × {SHL, SFOG} × {mobile, desktop}:
- Songbook chooser/switch
- Search: number/prefix, title, author, lyrics, fuzzy, case+order-insensitive, no-results, persistence, clear
- Infinite scroll (incl. tall-viewport case, #242)
- Lyric view, verse labels
- Music view + double-click zoom
- Second-tune toggle (SHL only; asserted **absent** for SFOG/others)
- Lyric↔music toggle
- Prev/next arrows (desktop), swipe nav (mobile, emulated touch)
- Download (music-mode only, correct name, png-not-html)
- Settings + dark-mode persistence across reload
- Feedback-form validation (**Octokit mocked — no real GitHub issues created**)
- Edge cases (bad songId in URL, songbook-fetch fallback)

### Verification
- Local run: **118 passed, 0 skipped** (6 suites)
- Runs in CI's existing **Build and Test** job on every PR (Node 16), enforced by branch protection
- eslint clean (0 errors), prettier clean
- Live data → added waits/retries for stability

### Notes
- Uses live data per request — no snapshot of the Resources repo committed here.
- Sampled (not full-sweep) per request; runs in the existing PR check, no separate nightly job.
- The old `App.test.tsx` (SHL-only, ~22 tests, desktop-only, duplicated/placeholder cases) is fully subsumed by the new suite.
- The 35 open issues are **backlogged** and will be cleaned up before any are worked; not treated as source of truth.

— The Bott

### CODE REVIEW

File: src/test/e2e/Layer1.data.test.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+// (156/216/278/478). SFOG mirrors the spread within its 280-song range.
+const SAMPLES: Record<BookId, number[]> = {
+  shl: [1, 6, 42, 44, 100, 156, 216, 250, 278, 333, 400, 478, 500, 533],
+  sfog: [1, 42, 44, 88, 100, 156, 200, 216, 250, 278, 280],
+};
+
+// Out-of-range edges per book: 0 and N+1.
+const EDGES: Record<BookId, number[]> = {
+  shl: [0, 534],
+  sfog: [0, 281],
```
Comment (MiTo0o) [inline src/test/e2e/Layer1.data.test.tsx]: maybe make the end value larger? Just in case sfog gets updated later you dont have to always update the n + 1 value. not a big deal

**Review by MiTo0o — DISMISSED** [review]
DEAL lgtm.
Just curious...... wondering why you went with Puppeteer?

File: src/test/e2e/Layer1.data.test.tsx (hunk 1 above)
Comment (brotatotes) [reply src/test/e2e/Layer1.data.test.tsx]: Good call — switched it to derive the high edge from live data at runtime (`max(songNumber)+1`) instead of the hardcoded n+1, so it self-updates if a songbook grows. Commit 8229353.

**Comment by brotatotes** [comment]
On Puppeteer: it was already the repo's pattern (old App.test.tsx used it), and these tests need a real browser — image loads (200/png), zoom width, scroll, mobile/desktop viewports — which jsdom can't do. Playwright would work too, just wasn't worth swapping the harness. Thanks for the review!

**Review by MiTo0o — APPROVED** [review]
🚀
